### PR TITLE
feat: make `@testing-library/dom` a peer dependency

### LINKED ISSUE #294: Dependency on `@testing-library/dom`
It's not apparent from the documentation why this plugin has a dependency on `@testing-library/dom`.

It looks like it does some branching based on whether a query is a DTL query or not but I can't find documentation how this affects rules in this repo.

This would help asses if we need a direct runtime dependency (current) or peer dependency or configuration option instead i.e. instead of requiring DTL and getting the list of queries, the user inputs that list. 

Documenting this internal behavior would help to understand if custom queries should also be considered.

Ideally there wouldn't be a dependency on /dom at all so that users just have to update /dom if that  releases a major and not update `/dom` and `eslint-plugin-jest-dom`. `jest-dom` doesn't have a dependency on `/dom` after all.

### PR DESCRIPTION
**Pull request by G-Rath**
**What**:

Makes `@testing-library/dom` a required peer dependency.

**Why**:

This ensures that the same version of `@testing-library/dom` that is being used on projects is used as part of linting, along with allowing us to support more versions, and just overall reduce downstream pain.

Also see #238 for prior discussion

**How**:

By making `@testing-library/dom` a peer dependency.

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged
      

Resolves #294
Resolves #301

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -22,8 +22,14 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        testing-library: [8, 9]
         eslint: [6.8.0, 6, 7.0.0, 7, 8.0.0, 8]
         node: [12.22.0, 12, 14.17.0, 14, 16.0.0, 16]
```
Comment (MichaelDeBoey) [inline .github/workflows/validate.yml:27]:
```suggestion
        eslint: [6.8.0, 6, 7.0.0, 7, 8.0.0, 8]
        node: [12.22.0, 12, 14.17.0, 14, 16.0.0, 16]
        testing-library-dom: [8, 9]
```

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -22,8 +22,14 @@ jobs:
...
+        exclude:
+          - testing-library: 9
+            node: 12.22.0
+          - testing-library: 9
+            node: 12
```
Comment (MichaelDeBoey) [inline .github/workflows/validate.yml:32]:
```suggestion
          - node: 12.22.0
            testing-library-dom: 9
          - node: 12
            testing-library-dom: 9
```

File: .github/workflows/validate.yml [hunk 3]
```diff
@@ -42,6 +48,9 @@ jobs:
       - name: Install ESLint v${{ matrix.eslint }}
         run: npm install --no-save --force eslint@${{ matrix.eslint }}
 
+      - name: Install @testing-library/dom v${{ matrix.testing-library }}
+        run: npm install --no-save --force @testing-library/dom@${{ matrix.testing-library }}
```
Comment (MichaelDeBoey) [inline .github/workflows/validate.yml:52]:
```suggestion
      - name: Install @testing-library/dom v${{ matrix.testing-library-dom }}
        run: npm install --no-save --force @testing-library/dom@${{ matrix.testing-library-dom }}
```

**Review by MichaelDeBoey — APPROVED** [review]
LGTM! 🚀

Renamed the variables to `testing-library-dom` for more clarity

**Comment by SimenB** [comment]
Awesome, thanks!

**Pull request merged**